ext/memcached_rb/memcached_rb.c

```diff
@@ -1,460 +1,518 @@
-  "AuthFailure", // MEMCACHED_AUTH_FAILURE
-  "AuthContinue", // MEMCACHED_AUTH_CONTINUE
-  "ParseError", // MEMCACHED_PARSE_ERROR
-  "ParseUserError", // MEMCACHED_PARSE_USER_ERROR
-  "Deprecated", // MEMCACHED_DEPRECATED
-  "InProgress", // MEMCACHED_IN_PROGRESS
-  "ServerTemporarilyDisabled", // MEMCACHED_SERVER_TEMPORARILY_DISABLED
-  "ServerMemoryAllocationFailure", // MEMCACHED_SERVER_MEMORY_ALLOCATION_FAILURE
-  "MaximumReturn", // MEMCACHED_MAXIMUM_RETURN
-  "ConnectionSocketCreateFailure", // MEMCACHED_CONNECTION_SOCKET_CREATE_FAILURE
+	NULL, // MEMCACHED_SUCCESS
+	"Failure", // MEMCACHED_FAILURE
+	"HostLookupFailure", // MEMCACHED_HOST_LOOKUP_FAILURE
+	"ConnectionFailure", // MEMCACHED_CONNECTION_FAILURE
+	"ConnectionBindFailure", // MEMCACHED_CONNECTION_BIND_FAILURE
+	"WriteFailure", // MEMCACHED_WRITE_FAILURE
+	"ReadFailure", // MEMCACHED_READ_FAILURE
+	"UnknownReadFailure", // MEMCACHED_UNKNOWN_READ_FAILURE
+	"ProtocolError", // MEMCACHED_PROTOCOL_ERROR
+	"ClientError", // MEMCACHED_CLIENT_ERROR
+	"ServerError", // MEMCACHED_SERVER_ERROR
+	"StdError", // MEMCACHED_ERROR
+	"DataExist", // MEMCACHED_DATA_EXISTS
+	"DataDoesNotExist", // MEMCACHED_DATA_DOES_NOT_EXIST
+	NULL, // MEMCACHED_NOTSTORED
+	NULL, // MEMCACHED_STORED
+	NULL, // MEMCACHED_NOTFOUND
+	"MemoryAllocationFailure", // MEMCACHED_MEMORY_ALLOCATION_FAILURE
+	"PartialRead", // MEMCACHED_PARTIAL_READ
+	"SomeError", // MEMCACHED_SOME_ERRORS
+	"NoServer", // MEMCACHED_NO_SERVERS
+	"End", // MEMCACHED_END
+	"Deleted", // MEMCACHED_DELETED
+	"Value", // MEMCACHED_VALUE
+	"Stat", // MEMCACHED_STAT
+	"Item", // MEMCACHED_ITEM
+	"Errno", // MEMCACHED_ERRNO
+	"FailUnixSocket", // MEMCACHED_FAIL_UNIX_SOCKET
+	"NotSupported", // MEMCACHED_NOT_SUPPORTED
+	"NoKeyProvided", // MEMCACHED_NO_KEY_PROVIDED
+	"FetchNotfinished", // MEMCACHED_FETCH_NOTFINISHED
+	"Timeout", // MEMCACHED_TIMEOUT
+	NULL, // MEMCACHED_BUFFERED
+	"BadKeyProvided", // MEMCACHED_BAD_KEY_PROVIDED
+	"InvalidHostProtocol", // MEMCACHED_INVALID_HOST_PROTOCOL
+	"ServerMarkedDead", // MEMCACHED_SERVER_MARKED_DEAD
+	"UnknownStatKey", // MEMCACHED_UNKNOWN_STAT_KEY
+	"E2big", // MEMCACHED_E2BIG
+	"InvalidArgument", // MEMCACHED_INVALID_ARGUMENTS
+	"KeyTooBig", // MEMCACHED_KEY_TOO_BIG
+	"AuthProblem", // MEMCACHED_AUTH_PROBLEM
+	"AuthFailure", // MEMCACHED_AUTH_FAILURE
+	"AuthContinue", // MEMCACHED_AUTH_CONTINUE
+	"ParseError", // MEMCACHED_PARSE_ERROR
+	"ParseUserError", // MEMCACHED_PARSE_USER_ERROR
+	"Deprecated", // MEMCACHED_DEPRECATED
+	"InProgress", // MEMCACHED_IN_PROGRESS
+	"ServerTemporarilyDisabled", // MEMCACHED_SERVER_TEMPORARILY_DISABLED
+	"ServerMemoryAllocationFailure", // MEMCACHED_SERVER_MEMORY_ALLOCATION_FAILURE
+	"MaximumReturn", // MEMCACHED_MAXIMUM_RETURN
+	"ConnectionSocketCreateFailure", // MEMCACHED_CONNECTION_SOCKET_CREATE_FAILURE
 };
-#define MEMCACHED_ERROR_COUNT 51
 
-VALUE rb_eMemcachedErrors[MEMCACHED_ERROR_COUNT];
+#define MEMCACHED_ERROR_COUNT (ARRAY_SIZE(MEMCACHED_ERROR_NAMES))
```

Comment:
It's good style to never hardcode the size of an array, because one day we'll add another element to it and our app will segfault, or stop working properly. I've replaced the hardcoded size with an `ARRAY_SIZE` macro.